transformer_engine/jax/csrc/modules.h

```diff
@@ -43,6 +43,13 @@ struct Shape {
     }
 };
 
+std::vector<size_t> MakeShapeVector(const NVTEShape shape) {
+    std::vector<size_t> shape_vector;
+    for (int i=0; i<shape.ndim; i++)
+        shape_vector.push_back(shape.data[i]);
+    return shape_vector;
+}
```

Comment:
I notices the `MakeShapeVector` is used to return a vector to `cpp_extensions.py`. We could make this lighter as returning the size of the shape instead of a vector

Reply:
We also have at least one workspace tensor (in layernorm backward) with different dimensions than the rest so vectorizing the shape information helps maintain a unified interface and makes the Python side a tad nicer. 

However, if you think there's a meaningful enough difference in making this lighter, we can certainly just return out the sizes directly.